core/server/worker/src/main/java/alluxio/worker/block/BlockHeartbeatReporter.java

```diff
@@ -62,12 +70,11 @@ public BlockHeartbeatReporter() {
   }
 
   /**
-   * Generates the report of the block store delta in the last heartbeat period. Calling this method
-   * marks the end of a period and the start of a new heartbeat period.
+   * Generates the report of the report and clear the states.
    *
    * @return the block store delta report for the last heartbeat period
    */
-  public BlockHeartbeatReport generateReport() {
+  public BlockHeartbeatReport generateReportAndClear() {
```

Comment:
this change is made to fix a existing bug in heartbeat threads.
if the report it generated but the heartbeat rpc fails, the block location info generated by the report will just be gone. so we added a revert method here as well as rename this function to generateReportAndClear to make it closer to its semantics.